packages/braintree-integration/src/braintree-fastlane/braintree-fastlane-payment-strategy.ts

```diff
@@ -199,20 +226,66 @@ export default class BraintreeFastlanePaymentStrategy implements PaymentStrategy
 
         const { getPaymentToken } = this.getBraintreeCardComponentOrThrow();
 
-        const { id } = await getPaymentToken({
+        const paymentToken = await getPaymentToken({
             name: { fullName },
             billingAddress: this.mapToPayPalAddress(billingAddress),
         });
 
+        const binGuest = paymentToken.paymentSource.card?.binDetails?.bin || '';
+        const threeDSGuest =
+            this.is3DSEnabled && threeDSFeature
+                ? await this.get3DS(paymentToken.id, binGuest)
+                : undefined;
+
         return {
             methodId,
             paymentData: {
                 deviceSessionId,
-                nonce: id,
+                nonce: threeDSGuest ?? paymentToken.id,
             },
         };
     }
 
+    /**
+     * 3DS
+     */
+    private async get3DS(nonce: string, bin: string): Promise<string> {
+        const state = this.paymentIntegrationService.getState();
+        const threeDSecure = await this.braintreeSdk.getBraintreeThreeDS();
+        const amount = state.getCartOrThrow().cartAmount;
+        const roundedAmount = amount.toFixed(2);
+
+        return new Promise<string>((resolve, reject) => {
+            threeDSecure.verifyCard(
+                {
+                    amount: +roundedAmount,
+                    nonce,
+                    bin,
+                    onLookupComplete: (_data, next) => {
+                        threeDSecure.on('customer-canceled', () => {
+                            reject(new PaymentMethodCancelledError());
+
+                            return this.onError && this.onError(new PaymentMethodCancelledError());
+                        });
+
+                        next();
+                    },
+                },
+                (
+                    verifyError: Braintree3DsVerifyCardError,
+                    payload: Braintree3DsVerifyCardPayload,
+                ) => {
+                    if (verifyError) {
+                        if (verifyError.code === 'THREEDS_VERIFY_CARD_CANCELED_BY_MERCHANT ') {
```

Comment:
Please, combine this two conditions into one

Reply:
fixed